src/hotspot/share/gc/shenandoah/c2/shenandoahSupport.cpp

```diff
@@ -1342,11 +1342,17 @@ void ShenandoahBarrierC2Support::pin_and_expand(PhaseIdealLoop* phase) {
     // Test for in-cset.
     // Wires !in_cset(obj) to slot 2 of region and phis
     Node* not_cset_ctrl = NULL;
-    test_in_cset(ctrl, not_cset_ctrl, val, raw_mem, phase);
+    if (!lrb->is_native()) {
```

Comment:
I think it deserves a comment why LRB-native is special.

Reply:
Yup, I added a comment there. Thanks!